src/xcode/ENA/ENA/Source/Services/Risk/Provider/__tests__/RiskProviderTests.swift

```diff
@@ -407,9 +407,82 @@ final class RiskProviderTests: XCTestCase {
 		XCTAssertFalse(store.shouldShowRiskStatusLoweredAlert)
 	}
 
+	// MARK: - RiskProvider stress test
+
+	func test_When_RequestRiskIsCalledFromDifferentThreads_Then_ItReturnsWithAlreadyRunningErrorOrCalculatedRisk() {
+
+		let numberOfRequestRiskCalls = 30
+		let numberOfExecuteENABackgroundTask = 10
+
+		let riskProvider = makeSomeRiskProvider()
+		let riskConsumer = RiskConsumer()
+		riskProvider.observeRisk(riskConsumer)
+
+		let didCallbackExpectation = expectation(description: "Called didCalculateRisk or didFailCalculateRisk.")
+		didCallbackExpectation.expectedFulfillmentCount = numberOfRequestRiskCalls + numberOfExecuteENABackgroundTask
+		riskConsumer.didCalculateRisk = { _ in
+			didCallbackExpectation.fulfill()
+		}
+
+		riskConsumer.didFailCalculateRisk = { error in
+			didCallbackExpectation.fulfill()
+
+			if error.isAlreadyRunningError {
+				return
+			}
+
+			XCTFail("Error besides of isAlreadyRunningError should not happen.")
+		}
+
+		let concurrentQueue = DispatchQueue(label: "RiskProviderStressTest", attributes: .concurrent)
+		for _ in 0...numberOfRequestRiskCalls - 1 {
+			concurrentQueue.async {
+				riskProvider.requestRisk(userInitiated: false)
+			}
+		}
+
+		let appDelegate = AppDelegate()
+		appDelegate.riskProvider = riskProvider
+
+		for _ in 0...numberOfExecuteENABackgroundTask - 1 {
+			appDelegate.executeENABackgroundTask { _ in }
+		}
+
+		waitForExpectations(timeout: 20)
```

Comment:
Can we use something like `.extraLong` here?

Reply:
changed